corehq/apps/accounting/static/accounting/js/pricing_table.js

```diff
@@ -1,30 +1,70 @@
+hqDefine('accounting/js/pricing_table', [
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/initial_page_data',
+    "hqwebapp/js/assert_properties",
+], function (
+    $,
+    ko,
+    _,
+    initialPageData,
+    assertProperties
+) {
+    var PricingTable = function (options) {
+        assertProperties.assert(options, [
+            'editions',
+            'planOptions',
+            'currentPlan',
+        ]);
+
         'use strict';
-        var self = {};
+        var self = this;
 
-        self.currentEdition = current_edition;
-        self.isRenewal = isRenewal;
-        self.editions = ko.observableArray(_.map(editions, function (edition) {
-            return pricingTableEditionModel(edition, self.currentEdition);
-        }));
+        self.currentPlan = ko.observable(options.currentPlan);
+        self.editions = options.editions;
 
-        self.selected_edition = ko.observable(isRenewal ? current_edition : false);
-        self.isSubmitVisible = ko.computed(function () {
-            if (isRenewal){
-                return true;
+        self.selectedPlan = ko.observable(options.currentPlan);
+
+        self.showMonthlyPricing = ko.observable(false);
+
+        self.refundCss = ko.computed(function () {
+            if (self.showMonthlyPricing()) {
+                return "hide-refund";
             }
-            return !! self.selected_edition() && !(self.selected_edition() === self.currentEdition);
+            return "";
         });
-        self.selectCurrentPlan = function () {
-            self.selected_edition(self.currentEdition);
+
+        self.isSubmitVisible = ko.computed(function () {
+            return !! self.selectedPlan() && !(self.selectedPlan() === self.currentPlan());
+        });
+
+        self.isCurrentPlanCommunity = ko.observable(options.currentPlan === 'community');
+
+        self.selectCommunityPlan = function () {
+            self.selectedPlan('community');
         };
 
-        self.form = undefined;
+        self.communityCss = ko.computed(function () {
+            if (self.selectedPlan() === 'community') {
+                return "selected-plan";
+            }
+            return "";
+        });
+
+        self.planOptions = ko.observableArray(_.map(options.planOptions, function (opt) {
+          return new PlanOption(opt, self);
+        }));
+
+        self.showNext = ko.computed(function () {
+            return self.selectedPlan() === 'community' || self.showMonthlyPricing();
+        });
+
         self.openDowngradeModal = function(pricingTable, e) {
-            var editionSlugs = _.map(self.editions(), function(e) { return e.slug(); });
-            self.form = $(e.currentTarget).closest("form");
-            if (editionSlugs.indexOf(self.selected_edition()) < editionSlugs.indexOf(self.currentEdition)) {
+            if (
+                self.editions.indexOf(self.selectedPlan()) <
```

Comment:
Seems like there's some overlap in this and the minimum subscription PR that could be pulled into a common file: determining what a downgrade is; the constants for plan names.